featurebyte/api/feature_list.py

```diff
@@ -719,15 +693,40 @@ def save(
         ... ], name="feature_lists_invoice_features")
         >>> feature_list.save()  # doctest: +SKIP
         """
+        self._check_object_not_been_saved(conflict_resolution=conflict_resolution)
+
+        feature_payloads = []
+        for feat in self.feature_objects.values():
+            feature_payloads.append(FeatureCreate(**feat.dict(by_alias=True)))
+
+        assert self.name is not None, "FeatureList name cannot be None"
+        feature_list_create = FeatureListCreateWithBatchFeatureCreation.create(
+            name=self.name,
+            features=feature_payloads,
+            conflict_resolution=conflict_resolution,
+            _id=self.id,
+        )
```

Comment:
If we create a feature list of 204 saved features & deleted it, it takes 14.5s for this new feature list saving flow (current flow in the main branch takes only 12.5s).
The main reason is that it takes more time to (new flow)
(1) it takes more times to combine 204 feature graph into a feature cluster graph
(2) validates feature cluster graphs
than making 204 local api calls (main branch flow).

Reply:
Combining list of feature graph into a feature clusters affects
(1) feature list document creation time (the most expensive part in create document creation 
https://github.com/featurebyte/featurebyte/blob/main/featurebyte/service/feature_list.py#L217-L223)
(2) the time taken to prepare batch feature cluster payload